### ISSUE TITLE
Support createdAt and updatedAt fields for database types

### ISSUE THREAD

**Issue by ramnivas** [exograph/exograph#907]
We already sort of support `createdAt` through default value, but we need a concept of `readonly` fields to make it semantically correct.

```exo
type Todo {
  ...
  createAt: Instant = now()
}
```

But here, API still allows specifying a new value. If we enable marking this field as `readonly` (or an equivalent syntax), this will make it sound.

For `updatedAt`, the same idea works as long as developers have defined a trigger. But we may want to support that more directly (by creating a trigger during schema generation).

Note that while the interceptor mechanism can also work, a trigger is a better solution in the environment where the database is accessed by clients other than an Exograph-based server.

**Comment by shiftlabs1** [comment]
I think if triggers can be "attributified", that will be nice . for e.g 
if i have a type Todo 
```
@access(true)
@before-trigger('name-of-function')
type Todo {
@pk id: Uuid = generate_uuid()
  description: String 
  createdAt: LocalDatetime = now()
updatedAt: LocalDateTime = now()
}
```
We can have the function already in the DB or define it as a postgres modeul type (this will also cater to views )  with defined paratemeter and return types if any . Such trigger function can then be shared across the schema

**Comment by ramnivas** [comment]
Fixed by #1097 

Moved @shiftlabs1's comment into a new issue #1099
